Fixing all served

alex-silverman opened the pull request:
Had to rework code to handle the fact that some of the selected comparison "variables" are actually comprised of multiple binary variables (e.g. Gender).

kiadso reviewed (changes requested):
hey I didn't have a chance to troubleshoot it but whenever I run Eva, upload the sample data, and select "Gender" and "Age" I get an error that says "object vars2 not found"

kiadso reviewed (commented):
whenever I run this and select Domestic Violence, it hangs and I get an error message saying there's no such column as DomesticViolence- thanks @alex-silverman!

alex-silverman commented:
@kiadso - I just pushed a number of changes to make sure things are working better and code is cleaner. 

I am noticing that the Hispanic-Focus Races/Ethnicities (when methodology type = Inclusive) + Gender is generating an error. I'll work on that tomorrow

alex-silverman commented:
@kiadso - okay can you try again? I've just fixed Hispanic-Focused Races/Ethnicities.

kiadso reviewed (changes requested):
- I'm not seeing the DV Fleeing category on the chart but I verified there are clients that should have that as their DomesticViolenceCategory.
- Also if you select Age and Gender, the Genders don't make sense. Like there's hardly any cis men/boys at all and there are a lot of "gender expansive, not transgender" clients.
- When I select Homelessness Type and any other selection, I don't see anything.
- When I select Age and Race/Ethnicities, the numbers also see incorrect, like the highest numbers are in Asian/Asian American Alone and Native Hawaiian/Pacific Islander.
- the numeric labels should use scales::comma(value) so they show with commas
- I really like the way the drop lists in the top section gray out when you go to this tab, it makes sense
- if I go to methodology 2 and select Hispanic-Focused Race/Ethnicities + Veteran Status or Veteran Status + All Races/Ethnicities, I get an error: [object Object]
- 
-

alex-silverman commented:
> * I'm not seeing the DV Fleeing category on the chart but I verified there are clients that should have that as their DomesticViolenceCategory.
> * Also if you select Age and Gender, the Genders don't make sense. Like there's hardly any cis men/boys at all and there are a lot of "gender expansive, not transgender" clients.
> * When I select Homelessness Type and any other selection, I don't see anything.
> * When I select Age and Race/Ethnicities, the numbers also seem incorrect, like the highest numbers are in Asian/Asian American Alone and Native Hawaiian/Pacific Islander.
> * the numeric labels should use scales::comma(value) so they show with commas
> * I really like the way the drop lists in the top section gray out when you go to this tab, it makes sense
> * if I go to methodology 2 and select Hispanic-Focused Race/Ethnicities + Veteran Status or Veteran Status + All Races/Ethnicities, I get an error: [object Object]
>   thanks for all your work on this!

thanks for these detailed notes. will take a look.

alex-silverman commented:
> * I'm not seeing the DV Fleeing category on the chart but I verified there are clients that should have that as their DomesticViolenceCategory.
> * Also if you select Age and Gender, the Genders don't make sense. Like there's hardly any cis men/boys at all and there are a lot of "gender expansive, not transgender" clients.
> * When I select Homelessness Type and any other selection, I don't see anything.
> * When I select Age and Race/Ethnicities, the numbers also seem incorrect, like the highest numbers are in Asian/Asian American Alone and Native Hawaiian/Pacific Islander.
> * the numeric labels should use scales::comma(value) so they show with commas
> * I really like the way the drop lists in the top section gray out when you go to this tab, it makes sense
> * if I go to methodology 2 and select Hispanic-Focused Race/Ethnicities + Veteran Status or Veteran Status + All Races/Ethnicities, I get an error: [object Object]
>   thanks for all your work on this!

Re #1, can you check again? I'm seeing The Currently Fleeing DV category:
![image](https://github.com/user-attachments/assets/7e8c8fd3-0004-411b-bfbd-9b5511d7d4f7)

Re #2 and #4, looks like the data became misaligned with the labels. Fixed!

Re #5, added commas!

Re #7, fixed this issue. Weirdly, it had to do with how we force the Race/Ethnciity selection to be the y-label axis. It's weird that the issue didn't show up before.

kiadso commented:
this is what I'm seeing for DV and Age (Not DV and Gender):
![image](https://github.com/user-attachments/assets/134258df-56ca-41f8-a2cb-0ee9d5ac0f48)
taking a look at the rest shortly!

kiadso commented:
Also I'm noticing the Gender and DV selections seem off as well, it has 53% of those who experienced DV as men, 21% as women- are we sure that's right?

alex-silverman commented:
> this is what I'm seeing for DV and Age (Not DV and Gender): ![image](https://private-user-images.githubusercontent.com/7561522/358291128-134258df-56ca-41f8-a2cb-0ee9d5ac0f48.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3NTg2MjUsIm5iZiI6MTcyMzc1ODMyNSwicGF0aCI6Ii83NTYxNTIyLzM1ODI5MTEyOC0xMzQyNThkZi01NmNhLTQxZjgtYTJjYi0wZWU5ZDVhYzBmNDgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTVUMjE0NTI1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzAxYmQ5YmYyN2ZiZTBmOTk1MGMxNDc5MmVkNDcxNmQ2NDIyZDgwMjdlYWNiNWY0YWU3YjdkMmRhMjQ2ZDA0NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.D-e1frk6HQjlGlMrlij23Ivob2LxQ4_oG1XRGCusb3g) taking a look at the rest shortly!

not able to replicate with Demo mode or loading hashed-current-good. Which file are you using?

alex-silverman commented:
@kiadso - ready for another review! I've incorporated fixes to the issues you mentioned. The data look more reasonable to me now. If you notice anything, please include the file used to test so I can replicate.

kiadso reviewed (commented):
- Whenever I go to Veteran Status and anything else (regardless of methodology), I don't see the Veteran cells with any numbers.
- in the detail box, it's showing Household Type and Level of Detail, which they may want to show conditionally
- Even with these issues, I'm not against merging them in so you can continue to work out the kinks. Let me know if you want to get this all the way there before merging or if you want to go ahead and merge and address any other issues in subsequent branches

alex-silverman commented:
> * Whenever I go to Veteran Status and anything else (regardless of methodology), I don't see the Veteran cells with any numbers.
> * in the detail box, it's showing Household Type and Level of Detail, which they may want to show conditionally
> * Even with these issues, I'm not against merging them in so you can continue to work out the kinks. Let me know if you want to get this all the way there before merging or if you want to go ahead and merge and address any other issues in subsequent branches

- good catch on Veteran Status. I guess I forgot to test that one! I had the underlying values as "Veteran" and "NonVeteran", but the variable is just a simple 1/0. I updated, and it's working now.
- I'll add the if statement now for the Household Type and Level of Detail. Easy enough to remove if they don't want to do that, and it's consistent with the Waterfall chart.

alex-silverman commented on server.R:
```diff
@@ -1007,7 +1014,10 @@ function(input, output, session) {
+    req(length(input$system_composition_filter) == 2)
+    syscomp_detailBox()
+  })
 
```
> I'm seeing duplicates: ![image](https://private-user-images.githubusercontent.com/7561522/359125150-60470c40-0594-4197-a9de-8af40697871c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQxMDI5MjYsIm5iZiI6MTcyNDEwMjYyNiwicGF0aCI6Ii83NTYxNTIyLzM1OTEyNTE1MC02MDQ3MGM0MC0wNTk0LTQxOTctYTlkZS04YWY0MDY5Nzg3MWMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTlUMjEyMzQ2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTBhN2Q1NTkwYmM0NTk4OGYyODMxNTRmMDkyY2NmMDhlZThkMWVlMzQ4NDU3MWI5MmMwNGFkYjM3MGVmYmFmYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.sz0K7gRc6beh03ehvKpMIA4sTBf4lxn0Qse-HFuYGGU)

I'm not sure why you're highlighting that particular chunk of code, but I found and fixed the duplicate, whicvh was around line 37 of the system_composition_functions.R script. I thought I'd already done this, but obviously not

alex-silverman commented:
@kiadso and @alex-silverman decided to merge this and fix any issues as hotfix branches
